Bottom: 3a41e425b3b018adb2b37ddaf3ad518089a80c26
Top:    f8f17b82c419faecd9b22becb86ff382f8f8f4a8
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-17 12:35:20 +0800

autofs-5.0.7 - use ulimit max open files if greater than internal maximum

When setting the maximum number of allowed file handles the current setting
should be checked before setting it. If the ulimit command has been used to
increase the maximum to larger than what automount would ask for then honour
it.


---

diff --git a/daemon/automount.c b/daemon/automount.c
index 019637f..1d0b64e 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2106,8 +2106,11 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
-	rlim.rlim_cur = MAX_OPEN_FILES;
-	rlim.rlim_max = MAX_OPEN_FILES;
+	res = getrlimit(RLIMIT_NOFILE, &rlim);
+	if (res == -1 || rlim.rlim_max <= MAX_OPEN_FILES)  {
+		rlim.rlim_cur = MAX_OPEN_FILES;
+		rlim.rlim_max = MAX_OPEN_FILES;
+	}
 	res = setrlimit(RLIMIT_NOFILE, &rlim);
 	if (res)
 		printf("%s: can't increase open file limit - continuing",
